src/results_analyze/analyzer_plugin.py

```diff
@@ -0,0 +1,133 @@
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+import attr
+
+from commoncode.cliutils import PluggableCommandLineOption
+from commoncode.cliutils import POST_SCAN_GROUP
+from plugincode.post_scan import PostScanPlugin
+from plugincode.post_scan import post_scan_impl
+
+from results_analyze import analyzer
+
+
+class NotAnalyzableResourceException(Exception):
+    pass
+
+
+@post_scan_impl
+class ResultsAnalyzer(PostScanPlugin):
+    """
+    Add the "license_detection_errors" list which has the license detection error
+    type information for each match errors.
+    """
+
+    resource_attributes = {
+        "license_detection_analysis": attr.ib(default=attr.Factory(list))
+    }
+
+    sort_order = 80
+
+    options = [
+        PluggableCommandLineOption(
+            ("--analyze-results",),
+            is_flag=True,
+            default=False,
+            help='Add the "license_detection_analysis" list which has the'
+            "license detection error type information for each match errors",
+            help_group=POST_SCAN_GROUP,
+        ),
+    ]
+
+    def is_enabled(self, analyze_results, **kwargs):
+        return analyze_results
+
+    def process_codebase(self, codebase, **kwargs):
+        for resource in codebase.walk():
+            if not resource.is_file:
+                continue
+
+            try:
+                # Will fail if missing attributes
+                if validate_resource(resource):
+                    resource.license_detection_analysis = analyze_resource(resource)
+                    codebase.save_resource(resource)
+            except NotAnalyzableResourceException as e:
+                msg = str(e)
+                codebase.errors.append(msg)
+                break
+
+
+def validate_resource(resource):
+    """
+    Return True if resource has all the data required for the analysis.
+    Return False if the resource does not have detected licenses.
+    Raise an exception if any of the essential attributes are missing from the resource.
+    """
+    has_licenses = hasattr(resource, "licenses")
+    licenses = getattr(resource, "licenses", [])
+    if has_licenses and not licenses:
```

Comment:
So these are different things, 

`has_licenses`:- True when the scan is run with --licenses option.
`not licenses`:- is also True when there weren't any licenses detected in that resource, i.e. `licenses` is an empty list. 

So both are required since we don't need to run the analysis step at all in the 2nd case of there not being any license detected in the resource. And `has_licenses` is required as we need to raise an error in the case that `--licenses` is not used, as it makes no sense to run the analysis without license detection. All these cases are tested in `test_analyzer_plugin.py`

Reply:
That True, False, Raise approach is confusing.
`validate_resource` should only validate the data content and you should implement a separate check for the field availability.

Reply:
Yes, changing this for a simpler approach. Thanks!